### PR TITLE
feat(wallet): Buy Deposit Buttons Asset Details

### PR DESCRIPTION
**Pull request by Douglashdaniel**
## Description 
Adds `Buy` and `Deposit` buttons to the `Asset Details` page if the asset is supported.

Resolves <https://github.com/brave/brave-browser/issues/26213>

## Submitter Checklist:

- [x] I confirm that no security/privacy review [is needed](https://github.com/brave/brave-browser/wiki/Security-reviews), or that I have [requested](https://github.com/brave/security/issues/new/choose) one
- [x] There is a [ticket](https://github.com/brave/brave-browser/issues) for my issue
- [x] Used Github [auto-closing keywords](https://docs.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue) in the PR description above
- [x] Wrote a good [PR/commit description](https://google.github.io/eng-practices/review/developer/cl-descriptions.html)
- [x] Squashed any review feedback or "fixup" commits before merge, so that history is a record of what happened in the repo, not your PR
- [x] Added appropriate labels (`QA/Yes` or `QA/No`; `release-notes/include` or `release-notes/exclude`; `OS/...`) to the associated issue
- [x] Checked the PR locally: `npm run test -- brave_browser_tests`, `npm run test -- brave_unit_tests`, `npm run lint`, `npm run gn_check`, `npm run tslint`
- [x] Ran `git rebase master` (if needed)

## Reviewer Checklist:

- [ ] A security review [is not needed](https://github.com/brave/brave-browser/wiki/Security-reviews), or a link to one is included in the PR description
- [ ] New files have MPL-2.0 license header
- [ ] Adequate test coverage exists to prevent regressions
- [ ] Major classes, functions and non-trivial code blocks are well-commented
- [ ] Changes in component dependencies are properly reflected in `gn`
- [ ] Code follows the [style guide](https://chromium.googlesource.com/chromium/src/+/HEAD/styleguide/c++/c++.md)
- [ ] Test plan is specified in PR before merging

## After-merge Checklist:

- [ ] The associated issue milestone is set to the smallest version that the
  changes has landed on
- [ ] All relevant documentation has been updated, for instance:
  - [ ] https://github.com/brave/brave-browser/wiki/Deviations-from-Chromium-(features-we-disable-or-remove)
  - [ ] https://github.com/brave/brave-browser/wiki/Proxy-redirected-URLs
  - [ ] https://github.com/brave/brave-browser/wiki/Fingerprinting-Protections
  - [ ] https://github.com/brave/brave-browser/wiki/Brave%E2%80%99s-Use-of-Referral-Codes
  - [ ] https://github.com/brave/brave-browser/wiki/Custom-Headers
  - [ ] https://github.com/brave/brave-browser/wiki/Web-Compatibility-Exceptions-in-Brave
  - [ ] https://github.com/brave/brave-browser/wiki/QA-Guide
  - [ ] https://github.com/brave/brave-browser/wiki/P3A

### Testing Fund Wallet
1. Go to the `Market` tab and select an asset.
2. After the `Asset Details` page loads, click on `Buy`.
3. You should be directed to the `Fund Wallet` page.

https://user-images.githubusercontent.com/40611140/197650475-9470752c-6806-4366-8503-81c8d33e9dc5.mov

### Testing Deposit Funds
1. Go to the `Market` tab and select an asset.
2. After the `Asset Details` page loads, click on `Deposit`.
3. You should be directed to the `Deposit Funds` page.

https://user-images.githubusercontent.com/40611140/197650532-25ede442-33e4-4ec8-a87e-71aa5af605ed.mov

### CODE REVIEW

File: components/brave_wallet_ui/components/desktop/views/portfolio/portfolio-asset.tsx [hunk 1]
```diff
@@ -340,6 +342,14 @@ export const PortfolioAsset = (props: Props) => {
 
   const isNftAsset = selectedAssetFromParams?.isErc721
 
+  const isBuySupported = React.useMemo(() => {
```
Comment (muliswilliam) [inline components/brave_wallet_ui/components/desktop/views/portfolio/portfolio-asset.tsx:345]: We should move `isBuySupported` and `isDepositSupported` to `useMultiChainBuyAssets` hook for better reuses

**Review by muliswilliam — CHANGES_REQUESTED** [review]
Just a small refactor.

**Review by muliswilliam — APPROVED** [review]
LGTM

**Comment by srirambv** [comment]
Verification passed on <html>
<body>


Brave | 1.47.91 Chromium: 108.0.5359.48 (Official Build) nightly (64-bit)
-- | --
Revision | `18ceeca0d99318e70c00d2e04d88aa55488b5c63-refs/branch-heads/5359@{#854}`
OS | Linux


</body>
</html>

- Verified steps from PR
- Verified Buy/Deposit buttons on asset details screen is shown
- Verified clicking Buy from asset details screen loads `Fund Wallet` page
- Verified clicking Deposit from asset details screen loads `Deposit Funds` page

https://user-images.githubusercontent.com/17010094/203355921-5b1b7e42-1c2d-453d-9f86-4eef7bf43790.mp4

**Pull request merged**